docs/codeql/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva.rst

```diff
@@ -0,0 +1,164 @@
+
+.. pull-quote::
+   
+   Note
+
+   - The controller repository visibility can be "public" if you plan to analyze public repositories. The variant analysis will be free.
+   - The controller repository visibility must be "private" if you need to analyze any private or internal repositories. Any actions minutes used by variant analysis, above the free limit, will be charged to the repository owner. For more information about free minutes and billing, see "`About billing for GitHub Actions <https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions>`__." 
+
+You must define a controller repository before you can run your first variant analysis.
+
+.. image:: ../images/codeql-for-visual-studio-code/controller-repository.png
+    :width: 350
+    :alt: Screenshot of the CodeQL extension in Visual Studio Code. The "Variant Analysis Repositories" section is expanded and the "Set up controller repository" button is highlighted with a dark orange outline.
+
+#. In Visual Studio Code, click **QL** in the left sidebar to display the CodeQL extension.
+
+#. Expand **Variant Analysis Repositories** and click **Set up controller repository** to display a field for the controller repository.
+
+#. Type the owner and name of the repository on GitHub.com that you want to use as your controller repository and press the **Enter** key.
+
+#. If you are prompted to authenticate with GitHub, follow the instructions and sign into your pesonal or organization account. When you have finished following the process, a prompt from GitHub Authentication may ask for permission to open a URI in Visual Studio Code, click **Open**.
+
+The name of the controller repository is saved in your settings for the CodeQL extension. For information on how to edit the controller repository, see ":ref:`Customizing settings <customizing-settings>`."
+
+Running a query at scale using variant analysis
+-----------------------------------------------
+
+#. Expand the **Variant Analysis Repositories** section, to show the default lists of the top 10, top 100, and top 1000 public repositories on GitHub.com. These are ranked by considering various metrics such as number of stars, number of watchers, number of forks etc.
+
+#. Select which GitHub repository or repositories you want to run your query against.
+
+   - Click **Top X repositories** to select this list of public repositories for analysis.
+   - Click **LIST-NAME** to select a custom list for analysis (see :ref:`later in this article <custom-lists>` for details).
+   - Click **Add new database**, the **+** icon, to add a repository or an organization to the panel, then click to select it for analysis.
```

Comment:
I don't understand the `then click to select it for analysis.` part of this sentence.

Reply:
Thanks. I'll rewrite this to make it clearer.